### PR TITLE
Fix NPM license URL

### PR DESCRIPTION
**Pull request by arcanis**
Old one was having issues with scopped packages, cf #757

### CODE REVIEW

**Comment by arcanis** [comment]
Ping @espadrine ?

**Comment by espadrine** [comment]
Is that enough? For instance, on your branch, the URL `/npm/l/@terse/webpack.svg` calls https://registry.npmjs.org/%40terse%2Fwebpack which returns 

```
{ code: 'ResourceNotFound',
  message: '/%40terse%2Fwebpack does not exist' }
```

The obvious change https://registry.npmjs.org/@terse/webpack returns `{"error":"Not found"}`.

**Comment by teppeis** [comment]
@espadrine The correct format is https://registry.npmjs.org/@terse%2fwebpack

**Comment by arcanis** [comment]
Looks fine - I went for the minimal required changes, but adding regex is another option :) Do you want me to update the PR?

**Comment by paulmelnikow** [comment]
Yes, if you don't mind. The version patch is now merged. Could you update this to match the way the version endpoint is handling URI components and request parameters?

**Comment by paulmelnikow** [comment]
This wasn't working for me as-is, so I updated it with a version which works on both scoped and unscoped packages, and incorporates the regex from #926.

@arcanis or @espadrine, would you like to give it a read?

File: server.js [hunk 1]
```diff
@@ -1673,21 +1673,39 @@ cache(function(data, match, sendBadge, request) {
 }));
 
 // npm license integration.
-camp.route(/^\/npm\/l\/(.*)\.(svg|png|gif|jpg|json)$/,
+camp.route(/^\/npm\/l\/(@[^\/]*)?\/?([^\/]*)\.(svg|png|gif|jpg|json)$/,
```
Comment (arcanis) [inline server.js]: Shouldn't the regexp be like this?

```
/^\/npm\/l\/(?:(@[^\/]+)\/)?([^\/]+)\.(svg|png|gif|jpg|json)$/
```

Comment (arcanis) [inline server.js]: Or better yet:

```
/^\/npm\/l\/(?:@([^\/]+)\/)?([^\/]+)\.(svg|png|gif|jpg|json)$/
```

This way, you can change the code below to:

```js
const pkg = encodeURIComponent(scope + '/' + repo);
apiUrl = `https://registry.npmjs.org/@${pkg}`;
```

By doing so, you would correctly escape both the scope and the package name

File: server.js [hunk 2]
```diff
@@ -1673,21 +1673,39 @@ cache(function(data, match, sendBadge, request) {
...
-  var badgeData = getBadgeData('license', data);
+  const scope = match[1];   // "@user"
+  const repo = match[2];    // "express"
+  const format = match[3];  // "svg"
+  let apiUrl;
+  if (scope === undefined) {
+    // e.g. https://registry.npmjs.org/express/latest
+    // Use this endpoint as an optimization. It covers the vast majority of
+    // these badges, and the response is smaller.
+    apiUrl = `https://registry.npmjs.org/${repo}/latest`;
```
Comment (arcanis) [inline server.js]: The repo isn't encoded here

**Comment by arcanis** [comment]
Sorry, busy month! I've left a few notes, I let you decide what to do with them :)

File: server.js (hunk 1 above)
Comment (paulmelnikow) [inline server.js]: Champion! That's some sweet regex-fu.

**Comment by paulmelnikow** [comment]
@arcanis Thanks, updated.

**Comment by paulmelnikow** [comment]
Since @arcanis and I have both worked on and read this, I thought I would go ahead and merge it.

**Pull request merged**